csrc/mamba/mamba_ssm/selective_scan_fwd.cu

```diff
@@ -108,17 +102,30 @@ void selective_scan_fwd_kernel(SSMParamsBase params) {
     const int batch_id = blockIdx.x;
     const int dim_id = blockIdx.y;
     const int group_id = dim_id / (params.dim_ngroups_ratio);
-    input_t *u = reinterpret_cast<input_t *>(params.u_ptr) + batch_id * params.u_batch_stride
+    int seqlen = params.seqlen;
+    int sequence_start_index = batch_id;
+    if constexpr (kVarlen){
+        int *cu_seq_len = reinterpret_cast<int *>(params.cu_seq_len_ptr);
+        sequence_start_index = batch_id == 0 ? 0 : cu_seq_len[batch_id - 1];
+        const int sequence_end_index = cu_seq_len[batch_id];
+        seqlen = sequence_end_index - sequence_start_index;
+    }
```

Comment:
similar question to above: Any load imbalance issues from parallelizing along batch when kVarlen is true?

Reply:
Yeah, same as mentioned above :)